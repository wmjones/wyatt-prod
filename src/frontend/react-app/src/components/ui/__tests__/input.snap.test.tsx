import React from 'react';
import renderer from 'react-test-renderer';
import { Input } from '../input';

describe('Input Component Snapshots', () => {
  it('renders default input correctly', () => {
    const tree = renderer
      .create(<Input placeholder="Default Input" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders retro input correctly', () => {
    const tree = renderer
      .create(<Input retro placeholder="Retro Input" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders different input types correctly', () => {
    const types = [
      'text',
      'password',
      'email',
      'number',
      'tel',
      'url',
      'search'
    ];

    types.forEach(type => {
      const tree = renderer
        .create(<Input type={type} placeholder={`${type} Input`} />)
        .toJSON();
      expect(tree).toMatchSnapshot(`${type} input`);
    });
  });

  it('renders disabled input correctly', () => {
    const tree = renderer
      .create(<Input disabled placeholder="Disabled Input" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders read-only input correctly', () => {
    const tree = renderer
      .create(<Input readOnly placeholder="ReadOnly Input" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders input with additional class name correctly', () => {
    const tree = renderer
      .create(<Input className="custom-class" placeholder="Custom Class Input" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
