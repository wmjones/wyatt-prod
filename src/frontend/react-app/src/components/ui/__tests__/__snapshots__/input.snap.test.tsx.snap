// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Component Snapshots renders default input correctly 1`] = `
<input
  className="flex h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-sm ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50"
  placeholder="Default Input"
/>
`;

exports[`Input Component Snapshots renders different input types correctly: email input 1`] = `
<input
  className="flex h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-sm ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50"
  placeholder="email Input"
  type="email"
/>
`;

exports[`Input Component Snapshots renders different input types correctly: number input 1`] = `
<input
  className="flex h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-sm ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50"
  placeholder="number Input"
  type="number"
/>
`;

exports[`Input Component Snapshots renders different input types correctly: password input 1`] = `
<input
  className="flex h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-sm ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50"
  placeholder="password Input"
  type="password"
/>
`;

exports[`Input Component Snapshots renders different input types correctly: search input 1`] = `
<input
  className="flex h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-sm ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50"
  placeholder="search Input"
  type="search"
/>
`;

exports[`Input Component Snapshots renders different input types correctly: tel input 1`] = `
<input
  className="flex h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-sm ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50"
  placeholder="tel Input"
  type="tel"
/>
`;

exports[`Input Component Snapshots renders different input types correctly: text input 1`] = `
<input
  className="flex h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-sm ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50"
  placeholder="text Input"
  type="text"
/>
`;

exports[`Input Component Snapshots renders different input types correctly: url input 1`] = `
<input
  className="flex h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-sm ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50"
  placeholder="url Input"
  type="url"
/>
`;

exports[`Input Component Snapshots renders disabled input correctly 1`] = `
<input
  className="flex h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-sm ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50"
  disabled={true}
  placeholder="Disabled Input"
/>
`;

exports[`Input Component Snapshots renders input with additional class name correctly 1`] = `
<input
  className="flex h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-sm ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 custom-class"
  placeholder="Custom Class Input"
/>
`;

exports[`Input Component Snapshots renders read-only input correctly 1`] = `
<input
  className="flex h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-sm ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50"
  placeholder="ReadOnly Input"
  readOnly={true}
/>
`;

exports[`Input Component Snapshots renders retro input correctly 1`] = `
<input
  className="flex h-10 w-full rounded-md text-sm ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 retro-input border-2 border-black px-4 py-2 bg-white focus:outline-none focus:ring-2 focus:ring-retro-purple"
  placeholder="Retro Input"
/>
`;
